### PR TITLE
feat: interactive mode

### PR DESCRIPTION
**Pull request by AlexsJones**
Closes # 

## ✅ Checks

- [x ] My pull request adheres to the code style of this project
- [ x] My code requires changes to the documentation
- [ x] I have updated the documentation as required
- [x ] All the tests have passed

### CODE REVIEW

File: cmd/analyze/analyze.go [hunk 1]
```diff
@@ -69,14 +74,36 @@ var AnalyzeCmd = &cobra.Command{
...
 		// print results
 		output, err := config.PrintOutput(output)
 		if err != nil {
 			color.Red("Error: %v", err)
 			os.Exit(1)
 		}
 		fmt.Println(string(output))
+
+		// Interactive mode
+		if interactiveMode && explain {
```
Comment (matthisholleville) [inline cmd/analyze/analyze.go:86]: Shouldnt we disallow the interactive mode with json output ?

**Review by matthisholleville — CHANGES_REQUESTED** [review]
Only one comment. Good job.

File: cmd/analyze/analyze.go [hunk 2]
```diff
@@ -69,14 +74,36 @@ var AnalyzeCmd = &cobra.Command{
...
-
 		// print results
 		output, err := config.PrintOutput(output)
 		if err != nil {
 			color.Red("Error: %v", err)
 			os.Exit(1)
 		}
 		fmt.Println(string(output))
+
+		// Interactive mode
```
Comment (bwplotka) [inline cmd/analyze/analyze.go:85]: Perhaps to trivial comment?

File: cmd/analyze/analyze.go [hunk 3]
```diff
@@ -69,14 +74,36 @@ var AnalyzeCmd = &cobra.Command{
...
+			agentClient := agent.NewAIAgent(config, output)
+
+			go agentClient.StartInteraction()
+			for {
+				select {
+				case res := <-sigs:
+					switch res {
+					default:
+						os.Exit(0)
+					}
```
Comment (bwplotka) [inline cmd/analyze/analyze.go:98]: just os.Exit(0) here would be the same?
```suggestion
					os.Exit(0)
```

File: cmd/analyze/analyze.go [hunk 4]
```diff
@@ -102,4 +129,6 @@ func init() {
 	AnalyzeCmd.Flags().IntVarP(&maxConcurrency, "max-concurrency", "m", 10, "Maximum number of concurrent requests to the Kubernetes API server")
 	// kubernetes doc flag
 	AnalyzeCmd.Flags().BoolVarP(&withDoc, "with-doc", "d", false, "Give me the official documentation of the involved field")
+	// interactive mode flag
+	AnalyzeCmd.Flags().BoolVarP(&interactiveMode, "interactive", "i", false, "Interactive mode to debug commands can only be used with --explain flag")
```
Comment (bwplotka) [inline cmd/analyze/analyze.go:133]:
```suggestion
	AnalyzeCmd.Flags().BoolVarP(&interactiveMode, "interactive", "i", false, "Enable interactive mode that allows further conversation with LLM about the problem. Works only with --explain flag")
```

**Review by bwplotka — APPROVED** [review]
👍🏽 Good start, thanks!

File: pkg/ai/agent/agent.go [hunk 5]
```diff
@@ -0,0 +1,66 @@
+package agent
```
[the remark this answers is not in the document]
Comment (AlexsJones) [reply pkg/ai/agent/agent.go:1]: how about we call it `interactions` package ?

File: cmd/analyze/analyze.go (hunk 3 above)
Comment (AlexsJones) [reply cmd/analyze/analyze.go:98]: Is that right? Wouldn't that make it exit 0 on any signal?

File: cmd/analyze/analyze.go (hunk 1 above)
Comment (AlexsJones) [reply cmd/analyze/analyze.go:86]: Good idea, less tokens

File: pkg/ai/interactive/interactive.go [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+import (
+	"fmt"
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/k8sgpt-ai/k8sgpt/pkg/analysis"
+	"github.com/pterm/pterm"
+)
+
+type AGENT_STATE int
```
Comment (matthisholleville) [inline pkg/ai/interactive/interactive.go:12]:
```suggestion
type INTERACTIVE_STATE int
```

File: pkg/ai/interactive/interactive.go [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+const (
+	prompt = "Given the following context: "
+)
+
+const (
+	E_RUNNING AGENT_STATE = iota
```
Comment (matthisholleville) [inline pkg/ai/interactive/interactive.go:19]:
```suggestion
	E_RUNNING INTERACTIVE_STATE = iota
```

File: pkg/ai/interactive/interactive.go [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+	E_EXITED              = iota
+)
+
+type InteractionRunner struct {
+	config        *analysis.Analysis
+	State         chan AGENT_STATE
```
Comment (matthisholleville) [inline pkg/ai/interactive/interactive.go:25]:
```suggestion
	State         chan INTERACTIVE_STATE
```

File: pkg/ai/interactive/interactive.go [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+	contextWindow []byte
+}
+
+func NewInteractionRunner(config *analysis.Analysis, contextWindow []byte) *InteractionRunner {
+	return &InteractionRunner{
+		config:        config,
+		contextWindow: contextWindow,
+		State:         make(chan AGENT_STATE),
```
Comment (matthisholleville) [inline pkg/ai/interactive/interactive.go:33]:
```suggestion
		State:         make(chan INTERACTIVE_STATE),
```

File: cmd/analyze/analyze.go [hunk 10]
```diff
@@ -69,19 +73,42 @@ var AnalyzeCmd = &cobra.Command{
...
-		fmt.Println(string(output))
+		fmt.Println(string(output_data))
+
+		if interactiveMode && explain {
+			if output == "json" {
+				color.Yellow("Caution: interactive mode using --json enabled may use additional tokens.")
+			}
+			sigs := make(chan os.Signal, 1)
+			signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+			agentClient := interactive.NewInteractionRunner(config, output_data)
```
Comment (matthisholleville) [inline cmd/analyze/analyze.go:90]:
```suggestion
			interactiveClient := interactive.NewInteractionRunner(config, output_data)
```

File: cmd/analyze/analyze.go [hunk 11]
```diff
@@ -69,19 +73,42 @@ var AnalyzeCmd = &cobra.Command{
...
+
+			go agentClient.StartInteraction()
```
Comment (matthisholleville) [inline cmd/analyze/analyze.go:92]:
```suggestion
			go interactiveClient.StartInteraction()
```

File: cmd/analyze/analyze.go [hunk 12]
```diff
@@ -69,19 +73,42 @@ var AnalyzeCmd = &cobra.Command{
...
+			for {
+				select {
+				case res := <-sigs:
+					switch res {
+					default:
+						os.Exit(0)
+					}
+				case res := <-agentClient.State:
```
Comment (matthisholleville) [inline cmd/analyze/analyze.go:100]:
```suggestion
				case res := <-interactiveClient.State:
```

**Review by matthisholleville — CHANGES_REQUESTED** [review]

**Review by matthisholleville — APPROVED** [review]
LGTM

**Pull request merged**